External music does not work anymore begins from 1.0.0 but works on 0.9.21

### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Linux

### Describe the bug

Good day, bad news.

External music does not work anymore begins from 1.0.0. I have Ubuntu 20.04.5 LTS (Focal Fossa). Build from source:

FOPT="-march=amdfam10 -O2"
cmake -B build -DUSE_SDL_VERSION=SDL2 -DENABLE_IMAGE=ON -DCMAKE_CXX_FLAGS="$FOPT"
cmake --build build

but in latest snapshot the same.

If I roll back git to 0.9.21 and build then external music works fine. Music in ./music/homm2_XX.ogg.
In 1.0.0 always play midi even if set external music.

In output I have:
[ERROR]	playSound:  Failed to create an audio chunk from memory. The error: Unsupported block alignment
but this error occurs in both 1.0.0 and 0.9.21. Only in 1.0.0 no external music.

Thanks.

### Save file

Save file not needed

### Additional info

_No response_

```
[ERROR] playSound: Failed to create an audio chunk from memory. The error: Unsupported block alignment
```

This error is returned by SDL when it tries to play the a-law or u-law compressed audio, and this audio has the wrong format or it is corrupted. This error is strange in itself (where did this corrupted x-law soundtrack come from?)

> ```
> [ERROR] playSound: Failed to create an audio chunk from memory. The error: Unsupported block alignment
> ```
> 
> This error is returned by SDL when it tries to play the a-law or u-law compressed audio, and this audio has the wrong format or it is corrupted. This error is strange in itself (where did this corrupted x-law soundtrack come from?)

$ file homm2_01.ogg
homm2_01.ogg: Ogg data, Vorbis audio, stereo, 44100 Hz, ~192000 bps

In 0.9.21 the same error, but external audio plays well

--- a/src/engine/system.cpp
+++ b/src/engine/system.cpp
@@ -390,8 +390,9 @@ bool System::GetCaseInsensitivePath( const std::string & path, std::string & cor
 {
     correctedPath.clear();
 
-    if ( path.empty() )
+    if ( path.empty() ) {
         return false;
+    }
 
     DIR * d;
     bool last = false;
@@ -399,10 +400,11 @@ bool System::GetCaseInsensitivePath( const std::string & path, std::string & cor
     const char * delimiter = "/";
 
     if ( path[0] == delimiter[0] ) {
+        correctedPath.append( delimiter );
+
         d = opendir( delimiter );
     }
     else {
-        correctedPath = curDir[0];
         d = opendir( curDir );
     }
 
@@ -417,7 +419,9 @@ bool System::GetCaseInsensitivePath( const std::string & path, std::string & cor
             return false;
         }
 
-        correctedPath.append( delimiter );
+        if ( subPathIter != splittedPath.begin() ) {
+            correctedPath.append( delimiter );
+        }
 
         // Avoid directory traversal and try to probe directory name directly.
         // Speeds up file lookup when intermediate directories have a lot of
@@ -464,8 +468,9 @@ bool System::GetCaseInsensitivePath( const std::string & path, std::string & cor
         }
     }
 
-    if ( d )
+    if ( d ) {
         closedir( d );
+    }
 
     return !last;
 }

